.bit/responses/2.3-Week 2 Step 3.md

```diff
@@ -1,48 +1,226 @@
+
+> API Documentation can be tricky sometimes...Here's something to [help](https://learn.parabola.io/docs/reading-api-docs)
+</details>
+
+:bulb: **Request Headers** tell the receiving end of the request what *type* of data is in the body.
+
 <details>
-<summary>❓ Where can I find the keys?</summary>
- </br>
+<summary>❓ How do I specify Request Headers?</summary>
+</br>
 
-* Navigate to the home page on the Micrsoft Azure portal (https://portal.azure.com/#home)
+Go back to the Face API documentation [here](https://westus.dev.cognitive.microsoft.com/docs/services/563879b61984550e40cbbe8d/operations/563879b61984550f30395236), and find the **Request headers** section. 
 
-<img width="1440" alt="Screen Shot 2021-02-04 at 4 00 33 PM" src="https://user-images.githubusercontent.com/28051494/106971033-edac0000-6702-11eb-8243-1b5c2318f76d.png">
+There are two headers that you need. I've provided the format below. Enter in the two header names and their two corresponding values. 
 
-* Click on the resource you need the keys for
+FYI: The `Content-Type`  header should be set to`'application/octet-stream'`.  This specifies a binary file.
 
-<img width="1438" alt="Screen Shot 2021-02-04 at 4 00 49 PM" src="https://user-images.githubusercontent.com/28051494/106971035-ef75c380-6702-11eb-965b-c3ef7b5a7574.png">
+```js
+    //COMPLETE THE CODE
+    let resp = await fetch(uriBase + '?' + params.toString(), {
+        method: '<METHOD>',  //WHAT TYPE OF REQUEST?
+        body: '<BODY>',  //WHAT ARE WE SENDING TO THE API?
+      
+      	//ADD YOUR TWO HEADERS HERE
+        headers: {
+            '<HEADER NAME>': '<HEADER VALUE>'
+        }
+    })
+```
 
-* On the left menu bar, locate the Resource Management section and click on "Keys and Endpoint"
+</details>
 
-<img width="1440" alt="Screen Shot 2021-02-04 at 12 26 36 PM" src="https://user-images.githubusercontent.com/28051494/106971042-f43a7780-6702-11eb-9e28-e6b2bc16fa22.png">
-<br>
+<details>
+<summary>❓ What is the URL?</summary>
+</br>
+
+Notice that the URL is just the uriBase with the params we specified earlier appended on.
+
+Fill in the `method`  and `body`.
+
+```js
+async function analyzeImage(img){
+    
+    const subscriptionKey = '<YOUR SUBSCRIPTION KEY>';
+    const uriBase = '<YOUR ENDPOINT>' + '/face/v1.0/detect';
+    let params = new URLSearchParams({
+        'returnFaceId': 'true',
+        'returnFaceAttributes': 'emotion'
+    })
+    //COMPLETE THE CODE
+    let resp = await fetch(uriBase + '?' + params.toString(), {
+        method: '<METHOD>',  //WHAT TYPE OF REQUEST?
+        body: '<BODY>',  //WHAT ARE WE SENDING TO THE API?
+        headers: {
+            '<HEADER NAME>': '<HEADER VALUE>'  //do this in the next section
+        }
+    })
+    let data = await resp.json();
+    
+    return data; 
+}
+```
 </details>
 
-## :camping: To move on, place your two secrets in the repository secrets: `API_ENDPOINT` AND `SUBSCRIPTION_KEY`
+</br>
+
+#### ⏬ Receiving Data
+<details>
```

Comment:
This section is kind of random, but I see how it's hard to incorporate it. Maybe we could have this outside a drop down since it doesn't really below anywhere else?